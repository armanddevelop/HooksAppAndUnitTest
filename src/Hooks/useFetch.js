import { useEffect, useRef, useState } from "react";
const initialState = {
  data: null,
  loading: true,
  error: null,
};
export const useFetch = (url) => {
  const [state, setState] = useState(initialState);
  const isMounted = useRef(true);
  useEffect(() => {
    return () => {
      isMounted.current = false;
    };
  }, []);
  useEffect(() => {
    setState(initialState);
    fetch(url)
      .then((resp) => resp.json())
      .then((data) => {
        //console.log(data);
        setTimeout(() => {
          if (isMounted.current) {
            setState({
              data,
              loading: false,
              error: null,
            });
          } else {
            console.log("no se llamo al setState");
          }
        });
      })
      .catch(() =>
        setState({ data: null, loading: false, error: "no fetch data" })
      );
  }, [url]);
  return {
    state,
  };
};
