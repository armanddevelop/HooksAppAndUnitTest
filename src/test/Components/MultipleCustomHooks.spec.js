import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import toJson from "enzyme-to-json";
import { MultipleCustomHooks } from "../../Components/examples/MultipleCustomHooks";
import { useFetch } from "../../Hooks/useFetch";
jest.mock("../../Hooks/useFetch");
describe("Test to MultipleCustomHooks", () => {
  it("SnapShot MultipleCustomHooks", () => {
    useFetch.mockReturnValue({
      data: null,
      loading: true,
      error: null,
    });
    const wrapper = shallow(<MultipleCustomHooks />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("Should show information", () => {
    useFetch.mockReturnValue({
      data: [
        {
          author: "licha",
          quote: "hola licha",
          quote_id: 1,
        },
      ],
      loading: false,
      error: null,
    });
    const wrapper = shallow(<MultipleCustomHooks />);
    //console.log("Esto vael wrapper ", wrapper.html());
    expect(wrapper.find(".alert").exists()).toBeFalsy();
    expect(wrapper.find(".mb-5").text().trim()).toBe("hola licha");
    expect(wrapper.find("footer").text().trim()).toBe("licha");
    expect(wrapper.find("button").exists()).toBeTruthy();
  });
});
