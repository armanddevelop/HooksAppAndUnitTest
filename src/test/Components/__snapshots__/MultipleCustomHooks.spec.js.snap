// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test to MultipleCustomHooks SnapShot MultipleCustomHooks 1`] = `
<Fragment>
  <h1>
    Breaking bad quotes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading..........
  </div>
</Fragment>
`;
